src/modules/Login/action.js

```diff
@@ -176,6 +190,53 @@ export const initializeAccount = (account: EdgeAccount, touchIdInfo: Object) =>
   }
 }
 
+export const mergeSettings = (
+  loadedSettings: Object,
+  defaults: Object,
+  types: Object
+): { finalSettings: Object, isOverwriteNeeded: boolean, isDefaultTypeIncorrect: boolean } => {
+  const finalSettings = {}
+  // begin process for repairing damaged settings data
+  let isOverwriteNeeded = false
+  let isDefaultTypeIncorrect = false
+  for (const key in defaults) {
+    // if the type of the setting default does not meet the enforced type
+    const defaultSettingType = typeof defaults[key]
+    if (defaultSettingType !== types[key]) {
+      isDefaultTypeIncorrect = true
+      console.error('MismatchedDefaultSettingType key: ', key, ' with defaultSettingType: ', defaultSettingType, ' and necessary type: ', types[key])
+    }
+    // if the type of the loaded setting does not meet the enforced type
+    // eslint-disable-next-line valid-typeof
```

Comment:
Why do you have the eslint rule exception?

Reply:
The eslint disabling is meant for the `typeof` operator a few lines down, I realize the commented-out console.log lines probably cause confusion (I will remove them). Eslint doesn't like us using the `typoef` operator to compare to anything other than a string (ie 'object' or 'string'), so comparing to a variable that holds the string value of 'object' or 'string' causes eslint to throw an error.